pkg/docker/docker.go

```diff
@@ -609,15 +610,30 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
+		removalDone <- struct{}{}
+	}
+	go func() {
+		sigchan := make(chan os.Signal, 1)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+		defer signal.Stop(sigchan)
+		defer close(sigchan)
+		select {
+		case <-sigchan:
+			removeContainer()
+			os.Exit(0)
```

Comment:
Why do we exit with code 0?

I'd expect the return code to be != 0. By default, the Go runtime exits with code 2 when I send a SIGINT to a running program.

I see that returning 0 is also what the code in https://github.com/kubernetes/kubernetes/tree/master/pkg/util/interrupt does, but I find it suspicious -- not something discussed under code review. All comments I see in the code review are related to documentation comments.